Mathlib/Topology/Defs/Basic.lean

```diff
@@ -195,6 +195,47 @@ scoped[Topology] notation (name := closure_of) "closure[" t "]" => @closure _ t
 scoped[Topology] notation (name := Continuous_of) "Continuous[" t₁ ", " t₂ "]" =>
   @Continuous _ _ t₁ t₂
 
+namespace TopologicalSpace
+open Topology Lean Meta PrettyPrinter.Delaborator SubExpr
+
+/-- Fails iff argument #`space` has a `TopologicalSpace` instance registered and
+that instance is defeq to argument #`inst`. -/
+def assertNonCanonical (space inst : ℕ) : DelabM Unit := do
+  let α ← withNaryArg space getExpr
+  let .some synthInst ← Meta.trySynthInstance (← Meta.mkAppM ``TopologicalSpace #[α]) | return ()
+  let inst ← withNaryArg inst getExpr
+  if ← Meta.isDefEq inst synthInst then failure
+
+/-- Delaborate unary notation referring to non-standard topologies. -/
+def delabUnary (mkStx : Term → DelabM Term) : Delab :=
+  withOverApp 2 <| whenPPOption Lean.getPPNotation do
+    assertNonCanonical 0 1 -- fall through to normal delab if canonical
+    let instD ← withNaryArg 1 delab
+    mkStx instD
```

Comment:
For the purpose of further extensibility (i.e. code reusing in other parts of the library), it might be worthwhile to consider not hardcoding the inputs `2` and `1` (in my new implementation there is no `0`).